packages/devtools_app/lib/src/app_size/app_size_screen.dart

```diff
@@ -93,6 +99,43 @@ class _AppSizeBodyState extends State<AppSizeBody>
     addAutoDisposeListener(_tabController);
   }
 
+  Future<void> maybeLoadAppSizeFiles() async {
+    final queryParams = loadQueryParams();
+    if (queryParams.containsKey(primaryAppSizeFilePropertyName)) {
+      preLoadingData = true;
+      final primaryAppSizeFile = await server.requestPrimaryAppSizeFile(
+          queryParams[primaryAppSizeFilePropertyName]);
+      DevToolsJsonFile secondaryAppSizeFile;
+      if (queryParams.containsKey(secondaryAppSizeFilePropertyName)) {
+        secondaryAppSizeFile = await server.requestSecondaryAppSizeFile(
+            queryParams[secondaryAppSizeFilePropertyName]);
+      }
+
+      if (primaryAppSizeFile != null) {
+        preLoadingData = true;
```

Comment:
should `preLoadingData = true;` be wrapped in setState?

Reply:
added a todo to check for this - going to land and polish in follow ups